src/io_source.rs

```diff
@@ -175,7 +175,7 @@ where
     }
 }
 
-#[cfg(target_os = "wasi")]
+#[cfg(all(target_os = "wasi", not(target_env = "p2")))]
```

Comment:
Can we do this?
```suggestion
#[cfg(all(target_os = "wasi", target_env = "p1"))]
```

Reply:
The reason I did it that way is because `not(target_env = "p2")` is backwards compatible with `rustc` versions prior to the introduction of the `wasm32-wasip2` target (i.e. all stable `rustc` versions as of this writing).  See [here](https://bytecodealliance.zulipchat.com/#narrow/stream/235408-rust-toolchain/topic/.5Bpost.5D.20Changes.20to.20Rust's.20WASI.20targets/near/468251805) for further discussion.  It _does_ mean we'll need to make changes when `p3` becomes available, though.  Happy to do whatever you think is best here.